src/components/app/ProfileViewer.js

```diff
@@ -53,6 +53,31 @@ type DispatchProps = {|
 type Props = ConnectedProps<{||}, StateProps, DispatchProps>;
 
 class ProfileViewer extends PureComponent<Props> {
+  manageBeforeUnloadListener() {
+    const { isUploading } = this.props;
+    if (isUploading) {
+      window.addEventListener('beforeunload', this.handleUnload);
+    } else {
+      window.removeEventListener('beforeunload', () => this.handleUnload);
```

Comment:
you'll need to use `this.handleUnload` instead of `() => this.handleUnload`:
```js
       window.removeEventListener('beforeunload', this.handleUnload);
```

The reason is that when you write `() => this.handleUnload`, this creates a new function object each time we run the line. Then the underlying system in the browser can't match this new function object with the one registered line 59, and as a result the listener is never removed.

I haven't checked, but I believe that because of this, the dialog will always appear after an upload.

Of course you'll need to do the same in `componentWillUnmount` below.